lambda_code/src/fhir_service.py

```diff
@@ -127,11 +134,28 @@ def search_immunizations(self, nhs_number: str, disease_type: str, params: str)
         return fhir_bundle
 
     @timed
-    def _validate_patient(self, imms: dict):
-        nhs_number = [x for x in imms["contained"] if x.get("resourceType") == "Patient"][0]["identifier"][0]["value"]
+    def _validate_patient(self, imms: dict) -> dict:
+        """
+        Get the NHS number from the contained Patient resource and validate it with PDS.
+
+        If the NHS number doesn't exist, check the verification status code to make sure it's "04".
+        If the verification status code is "04", return an empty dict. If the NHS number exists, get the patient details
+        from PDS and return the patient details.
+        """
+        try:
+            nhs_number = [x for x in imms["contained"] if x["resourceType"] == "Patient"][0]["identifier"][0]["value"]
+        except (KeyError, IndexError):
+            nhs_number = None
+
+        if not nhs_number:
```

Comment:
This logic (lines 151, 153 and 154) would be better in post validations. Does it cause any problem to delete it from here?

Reply:
done